### PR TITLE
Tilepy space

### PR DESCRIPTION
**Pull request by hashkar**
This is the first PR to implement scheduling with space observatories. The strategy is now implemented for 2D Nobs scheduling only (PGWinFoV_NObs). The user can try scheduling with different observatories using the scripts launcher/Tiling_Observations_Nobs_Space.ipynb and launcher/Tiling_Observations_Nobs_Space.py and can try mixing between ground-based and space-based observatories. 
The observatories that are currently handled ad the ones that can be found in 'https://celestrak.com/NORAD/elements/science.txt'. Swift is included.

### CODE REVIEW

File: src/tilepy/include/PointingTools.py [hunk 1]
```diff
@@ -1607,7 +1617,134 @@ def ZenithAngleCut(prob, nside, time, minProbcut, maxZenith, observatory, minMoo
     return ObsBool, yprob
 
 
-def ComputeProbability2D(prob, highres, radecs, reducedNside, HRnside, minProbcut, time, observatory, maxZenith, FOV, tname, ipixlist, ipixlistHR, counter, dirName, useGreytime, plot):
+def GetSatelliteName(satellitename):
```
Comment (mseglar) [inline src/tilepy/include/PointingTools.py:1620]: please no hardcoded txt in the code... you can use this function to pre-launching tilepy, to get the coordinates/name. The name of the satellite should be in the config anyways.

File: src/tilepy/include/PointingTools.py [hunk 2]
```diff
@@ -1727,11 +1870,25 @@ def ComputeProbability2D(prob, highres, radecs, reducedNside, HRnside, minProbcu
...
             tempmask2 = separations > (radius - 0.05 * radius) * u.deg
+            hp.visufunc.projplot(skycoord[tempmask & tempmask2].ra, skycoord[tempmask &
+                                 tempmask2].dec, 'r.', lonlat=True, coord="C", linewidth=0.1)
             hp.visufunc.projplot(skycoord[tempmask & tempmask2].ra, skycoord[tempmask &
                                  tempmask2].dec, 'r.', lonlat=True, coord="C", linewidth=0.1)
             altcoord = np.empty(1000)
             azcoord = np.random.rand(1000) * 360
 
+            if ipixlistOcc == None:
+                print("No pcculted pix")
```
Comment (mseglar) [inline src/tilepy/include/PointingTools.py:1881]: typo. Also there that if it is just for testing. In general you would like to not have a print and instead only have:
if ipixlistOcc!=None:

File: examples/config/FollowupParameters_SWIFT.ini [hunk 3]
```diff
@@ -0,0 +1,34 @@
+[observatory]
+base = space
+name = SWIFT
```
Comment (mseglar) [inline examples/config/FollowupParameters_SWIFT.ini:3]: It is not SWIFT, but Swift

File: src/tilepy/include/PointingTools.py [hunk 4]
```diff
@@ -40,7 +40,7 @@
 import six
 import tables
 from astropy import units as u
-from astropy.coordinates import EarthLocation, get_body
+from astropy.coordinates import EarthLocation, get_body, get_sun, get_moon
```
[the remark this answers is not in the document]
Comment (hashkar) [reply src/tilepy/include/PointingTools.py:43]: will remove them they are not used anyway

**Review by mseglar — COMMENTED** [review]
Find here somments to the code

File: examples/config/FollowupParameters_SWIFT.ini (hunk 3 above)
Comment (hashkar) [reply examples/config/FollowupParameters_SWIFT.ini:3]: No it will not work as the database needs the name of the observatory in capital letters

**Comment by mseglar** [comment]
Did you test it in the single observatory case?

File: src/tilepy/include/PointingTools.py (hunk 2 above)
Comment (hashkar) [reply src/tilepy/include/PointingTools.py]: fixed

**Comment by hashkar** [comment]
> Find here comments to the code

I don't understand

**Comment by hashkar** [comment]
> Did you test it in the single observatory case?

yes

File: src/tilepy/include/PointingTools.py (hunk 1 above)
Comment (hashkar) [reply src/tilepy/include/PointingTools.py]: resolved

**Pull request merged**